### ISSUE TITLE
Running questions from the past

### ISSUE THREAD

**Issue by maksymilian-dabkowski** [BrownCLPS/LingView#16]
Below are some questions from the past. I am not sure how relevant they are at this point.

1. Are the XML files generated by ELAN sufficient to form a fast, searchable
database of texts? Although we normally access a single XML file at a time, building a
concordance will require searching through multiple texts. Generally, it seems like XML
shouldn’t be used as a database ( SE ), mostly due to speed concerns. However, the
small number of texts means that searching will still probably be quite fast. Is it therefore
acceptable to just search through multiple XML files? Is this something we should worry
about if we want the code to be useable in other projects, too?

2. Can the concordance and other features run offline, without using a server? We
could simply use a Python SimpleHTTPServer or something similar, e.g., XAMPP for an
Apache server, but this is complicated to use, doesn’t work well on mobile, and even if it
does work on mobile may have issues with different hardware and operating systems.

3. Should we use an XML or JSON format for data storage? XML is the format used by
ELAN and ETST and therefore preferred by default, while JSON behaves well with
modern JS frameworks and is used in the Moro database, which implements searching
on a very lightweight server. This question is just about data storage on the server; we
should use XML when archiving.

4. What ELAN tiers should we use, and how to those affect concordance, glossing,
etc.? This is especially important when we need to find corresponding tiers to display at
the same time, which is currently done via ANNOTATION_REF identifiers. We need to
make this decision soon, because tier names will affect our transcription process. We
also need to be prepared for potentially missing or incomplete tiers.

**Comment by sciepsilon** [comment]
1. Not relevant. We're (I believe) searching over our own generated json files, relying partly on a search_index.json file that we build for this purpose.
2. Maybe it does already. I'd have to look at the search code to know for sure. 
3. We chose JSON. We're not likely to switch to XML any time soon.
4. We chose to accept any and all ELAN tiers. This mostly works well, but there are certain features (Search/concordance, and maybe saving user preferences for what tiers to display) that would work much better if we restricted the available tiers. I don't think we will though.
